extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -0,0 +1,232 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Layout} from '../../../src/layout';
+import {log} from '../../../src/log';
+import {loadPromise} from '../../../src/event-helper';
+import {mergeObjects} from '../../../third_party/data-layer-helper/helper';
+import {urlReplacementsFor} from '../../../src/url-replacements';
+
+import {addListener} from './instrumentation';
+import {AnalyticsConfig} from './vendors';
+
+export class AmpAnalytics extends AMP.BaseElement {
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return true;
+  }
+
+  /** @override */
+  buildCallback() {
+
+    /**
+     * @private {?JSONValue} The analytics config associated with the tag
+     */
+    this.config_ = null;
```

Comment:
It's used everywhere as a non-nullable value. Can you redefine this code to make it clear and change typing? Also, I think it should be `!JSONObject`